lib/aba/batch.rb

```diff
@@ -47,56 +47,68 @@ def initialize(attrs = {}, transactions = [])
     end
 
     def to_s
-      raise RuntimeError, 'No transactions present - add one using `add_transaction`' if @transactions.empty?
+      raise RuntimeError, 'No entries present - add one using `add_transaction` or `add_return`' if entries.empty?
       raise RuntimeError, 'ABA data is invalid - check the contents of `errors`' unless valid?
 
       # Descriptive record
       output = "#{descriptive_record}\r\n"
 
       # Transactions records
-      output += @transactions.map{ |t| t[1].to_s }.join("\r\n")
+      output += entries.map { |t| t[1].to_s }.join("\r\n")
 
       # Batch control record
       output += "\r\n#{batch_control_record}"
     end
 
     def add_transaction(attrs = {})
-      if attrs.instance_of?(Aba::Transaction)
-        transaction = attrs
-      else
-        transaction = Aba::Transaction.new(attrs)
-      end
-      @transactions[@transaction_index] = transaction
-      @transaction_index += 1
-      transaction
+      add_entry(Aba::Transaction, attrs)
+    end
+
+    def add_return(attrs = {})
+      add_entry(Aba::Return, attrs)
+    end
+
+    def transactions
+      entries.select { |entry| entry.instance_of?(Aba::Transaction) }
     end
 
     def transactions_valid?
-      !has_transaction_errors?
+      !transactions.map { |t| t[1].valid? }.include?(false)
+    end
+
+    def entries_valid?
+      !has_entry_errors?
```

Comment:
Sure. The reason for adding it is because there is a public `transactions_valid?`. That's not as useful anymore because it only covers transactions but needs to stick around for backward compatibility. I wanted to provide a more useful alternative.

I'm happy to remove it `entries_valid?` if you'd prefer.